Sources/_StringProcessing/RegexDSL/Core.swift

```diff
@@ -12,9 +12,11 @@ public protocol RegexProtocol {
 }
 
 /// A regular expression.
+@frozen
```

Comment:
What's the rationale for this? I was expecting `Regex` to be resilient, and certainly not expecting to put the program class in our ABI

Reply:
When do you expect it to be resilient?  `Program` acts like the guts that caches the compiled program, and I think that expansion to `Regex` could be added to that type.  As mentioned in the PR description, I'd like to avoid it address-only type, but I can change if we need to.

Reply:
I might be prematurely optimizing this, but I was thinking that if we froze the-type-that-can-be-executed, we'd boil it down to a `_NativeObject`-like reference to supports immortal references for literals. 

(Also, I don't know if we'll use the same type for a linked/executable program as the type used for program composition)

What are you worried about specifically if this is address only? Is there something you want or need to inline elsewhere to get around the generic parameters?

Reply:
> What are you worried about specifically if this is address only? Is there something you want or need to inline elsewhere to get around the generic parameters?

Just slightly more straightforward to emit the code in SILGen as well as file-checking it. But sure, let me change it back to non-frozen.